apps/06.1Checkers.js

```diff
@@ -53,15 +92,46 @@ function Board() {
   };
 
   // Your code here
+  this.selectChecker = function(row, column) {
+    return this.grid[row][column];
+  }
+
+  this.killChecker = function(position) {
+    var checkerPosition = this.selectChecker(position[0], position[1]);
+    var find = this.checkers.indexOf(checkerPosition);
+    this.checkers.splice(find, 1);
+    this.grid[position[0]][position[1]] = null;
+  }
 }
+
 function Game() {
 
   this.board = new Board();
 
   this.start = function() {
     this.board.createGrid();
-    // Your code here
+    this.board.createCheckers();
   };
+
+  this.moveChecker = function(start, end) {
+    var startRow = Number(start[0]);
+    var startColumn = Number(start[1]);
+    var endRow = Number(end[0]);
+    var endColumn = Number(end[1]);
+    var checker = this.board.selectChecker(startRow, startColumn);
+    this.board.grid[startRow][startColumn] = null;
+    this.board.grid[endRow][endColumn] = checker;
+    
+    var distance = Math.abs(startRow - endRow);
+    var midRow = (startRow + endRow) / 2;
+    var midCol = (startColumn + endColumn) / 2;
+    var killPosition = [midRow, midCol];
+
+    if(distance === 2) {
+      this.board.killChecker(killPosition);
```

Comment:
Nice work on this app!

![success](https://media.giphy.com/media/26gsobowozGM9umBi/giphy.gif)